### DISCUSSION TITLE
Way to avoid data loss with QFieldCloud ?

### DISCUSSION THREAD

**Discussion by marius851000** [opengisch/QField#2760] [Q&A]
Hello. I tried to use QField for mapping street light for OpenStreetMap. After learning the basis of QGIS and QField, I made a project, with already mapped street light and streets (geopackage). I uploaded it to QField on my phone via QFieldCloud, and started mapping some street lights, marking streets with a street light mapped as such.

Back home, I pressed the "push changes" button, which the phone seemingly performed correctly (button greyed after some loading). However, in QGis, when trying to import changes, it said there weren't any changes (the data on my phone were still up-to-date). After trying to find a way to export the project (but it didn't seems to be on the shared accessible folder, and I weren't able to find were cloud project were stored on the phone, even with rooted ghost commander), I pressed "synchronise" button, which put on my phone the same thing that were on QGis (and removed all local changes).

I would like to ask if there are way to avoid such things in the future. (also act as an implicit feature request to take more care about not deleting data when doing syncronisation, like for example asking before overwriting file, or being able to export a cloud project).

(ps: I have been unable to reproduce the issue, and changes included both modifying existing way and node, as well as adding nodes)

**Comment by suricactus** [comment]
@marius851000 thanks for testing QFieldCloud during the beta phase. 

Let's start with the good news: your work is not lost! However, there is a problem to synchronize your data with your mother project and I am currenctly investigation where is the problem. You can easily see this when you go to the web interface of QFieldCloud, go to the projects and check the "changes" page. You will see all your changes as "pending" and suggestion to apply the changes. However, I would recommend you to wait a bit before hitting the button and having the clean picture what was going on.

I will keep you posted.

**Reply by suricactus** [reply]
I post the link of another related issue https://github.com/opengisch/qfieldcloud/issues/314 .

**Comment by suricactus** [comment]
Please follow the update on https://github.com/opengisch/qfieldcloud/issues/314 . All should be good now, no data loss expected.

**Reply by marius851000** [reply]
Thank. It worked. Have a nice day, and on my side, I'll probably return map some street lamp.

**Reply by suricactus** [reply]
Thanks for being proactive with QFieldCloud and bringing our attention. And also - great project!
